Core/src/TrackFitting/GlobalChiSquareFitter.cpp

```diff
@@ -13,15 +13,16 @@
 namespace Acts::Experimental {
 
 BoundVector calculateDeltaParams(const BoundMatrix& aMatrix,
-                                 const BoundVector& bVector) {
+                                 const BoundVector& bVector,
+                                 const std::size_t ndfSystem) {
   BoundVector deltaParams = BoundVector::Zero();
-  if (aMatrix(4, 4) == 0) {
+  if (ndfSystem == 4) {
     constexpr std::size_t reducedMatrixSize = 4;
     deltaParams.topLeftCorner<reducedMatrixSize, 1>() =
         aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
             .colPivHouseholderQr()
             .solve(bVector.topLeftCorner<reducedMatrixSize, 1>());
-  } else if (aMatrix(5, 5) == 0) {
+  } else if (ndfSystem == 5) {
     constexpr std::size_t reducedMatrixSize = 5;
     deltaParams.topLeftCorner<reducedMatrixSize, 1>() =
         aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
```

Comment:
I wonder if this has an advantage over slicing the matrices with the runtime `ndfSystem`. Another solution might be using one of the template unrolling mechanisms we have so you can deduplicate the code.

Ultimately this is also up to what performs better CPU time wise.

Reply:
`safeInverse()` needs to know the matrix dimensions at compile time.

For the unrolling, I wanted to wait until the material is ready, because there we need to unroll one of them to around 100, but I wanted to keep it like this for better probing the function.

Reply:
Right but nothing holds us back from having a runtime impl for `saveInverse`